# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/*/mtools/hotfixes.patch
# Copyright (C) 1998 - 2003 ROCK Linux Project
# 
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version. A copy of the GNU General Public
# License can be found at Documentation/COPYING.
# 
# Many people helped and are helping developing ROCK Linux. Please
# have a look at http://www.rocklinux.org/ and the Documentation/TEAM
# file for details.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

--- ./devices.h.orig	Sun Mar 12 17:49:46 2000
+++ ./devices.h	Sun Mar 12 17:50:06 2000
@@ -13,7 +13,12 @@
 #endif  /* MINOR not defined */
 
 #else
  
+#define _LINUX_STRING_H_ 
+#define __KERNEL__
+#  include <linux/kdev_t.h>
+#  include <linux/list.h>
+#undef __KERNEL__
 #include <linux/fs.h>        /* get MAJOR/MINOR from Linux kernel */
 #ifndef major
 #define major(x) MAJOR(x)
--- ./mformat.c.orig	Sun Mar 12 17:49:46 2000
+++ ./mformat.c	Sun Mar 12 17:50:27 2000
@@ -24,6 +24,11 @@
 #endif
 
 #ifdef OS_linux
+#define _LINUX_STRING_H_
+#define __KERNEL__
+#  include <linux/kdev_t.h>
+#  include <linux/list.h>
+#undef __KERNEL__
 #include "linux/hdreg.h"
 #include "linux/fs.h"
 #endif
--- ./mpartition.c.orig	Sun Mar 12 17:49:46 2000
+++ ./mpartition.c	Sun Mar 12 17:49:57 2000
@@ -16,6 +16,11 @@
 #include "partition.h"
 
 #ifdef OS_linux
+#define _LINUX_STRING_H_
+#define __KERNEL__
+#  include <linux/kdev_t.h>
+#  include <linux/list.h>
+#undef __KERNEL__
 #include "linux/hdreg.h"
 #include "linux/fs.h"
 #endif
--- ./mzip.c.orig	Sun Mar 12 17:49:46 2000
+++ ./mzip.c	Sun Mar 12 17:50:18 2000
@@ -30,6 +30,11 @@
 #include <sys/mount.h>
 #else
 #define _LINUX_KDEV_T_H 1  /* don't redefine MAJOR/MINOR */
+#define _LINUX_STRING_H_
+#define __KERNEL__
+#  include <linux/kdev_t.h>
+#  include <linux/list.h>
+#undef __KERNEL__
 #include <linux/fs.h>
 #endif
 
